crates/core/c8y_translator/src/smartrest.rs

```diff
@@ -0,0 +1,148 @@
+use chrono::prelude::*;
+use clock::{Clock, WallClock};
+use serde::{Deserialize, Serialize};
+use serde_json;
+
+use crate::json::CumulocityJsonError;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ThinEdgeJsonAlarm {
+    pub message: Option<String>,
+    pub time: Option<DateTime<FixedOffset>>,
+}
```

Comment:
This struct feels incomplete, ignoring the key attributes of an alarm.

Why not something along the following lines to separate the keys from the content of the alarm.
The former will be extract from the topic and the latter deserialized from the payload.

```suggestion
pub struct ThinEdgeJsonAlarm {
    pub name: String,
    pub severity: Severity,
    pub event: ThinEdgeJsonAlarmEvent,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ThinEdgeJsonAlarmEvent {
    pub message: Option<String>,
    pub time: Option<DateTime<FixedOffset>>,
}
```

Reply:
Oh. I just saw `crates/core/thin_edge_json/src/alarm.rs`.

The structs that are defined there must be used here. The deserialization from thin-edge-json must be move there.
And only the smartrest serialization should stay here.